app/renderer/js/main.js

```diff
@@ -432,6 +434,13 @@ class ServerManagerView {
 		} catch (err) {
 		}
 
+		if (ConfigUtil.getConfigItem('useServerLanguage', false)) {
+			const newLanguage = DomainUtil.getServerLanguage(this.tabs[index].webview.props.url);
+			if (newLanguage) {
+				ConfigUtil.setConfigItem('spellcheckerLanguage', newLanguage);
```

Comment:
`activateTab` is called multiple times throughout the app - so does for everytime we would be updating `spellcheckerLanguage` with the domains language? Is this necessary?

Reply:
Well, we need to know in `spellchecker.js` which language to switch to every time the tab is switched. We could also update something like `currentTabIndex` and then get the server language using that in the spellchecker file, but that wouldn't be any different. 

If this setting of org level spellchecker is set, we'll have to change language on a tab switch - I don't see a way around that. (Would love to be told otherwise though).

Reply:
Can't we get the server language from preload.js using `page_params.default_language` ?

Reply:
Yeah, that's a good idea. I just checked and the spellchecker works fine with the tab switch with just using `preload.js`. 

Looks like I'll have to refactor this a fair bit, but this should fix Abhigyan's concerns in https://github.com/zulip/zulip-desktop/pull/752#issuecomment-500694499